spikeinterface/core/addtemplatesrecording.py

```diff
@@ -0,0 +1,129 @@
+            self.add_recording_segment(recording_segment)
+
+
+
+class AddTemplatesRecordingSegment(BaseRecordingSegment):
+
+    def __init__(self, sampling_frequency: float, t_start: int, dtype, spike_vector: np.ndarray, templates: np.ndarray, nbefore: List[int],
+                 amplitude_factor: List[List[float]], parent_recording_segment: Union[BaseRecordingSegment, None] = None, t_max: Union[int, None] = None) -> None:
+
+        BaseRecordingSegment.__init__(self, sampling_frequency, t_start=t_start)
+        assert not (parent_recording_segment is None and t_max is None)
+
+        self.dtype = dtype
+        self.spike_vector = spike_vector
+        self.templates = templates
+        self.nbefore = nbefore
+        self.amplitude_factor = amplitude_factor
+        self.parent_recording = parent_recording_segment
+        self.t_max = parent_recording_segment.get_num_frames() if t_max is None else t_max
+
+    def get_traces(self, start_frame: Union[int, None] = None, end_frame: Union[int, None] = None,
+                   channel_indices: Union[List, None] = None) -> np.ndarray:
+        n_channels = self.templates.shape[2]
+        start_frame = 0 if start_frame is None else start_frame
+        end_frame = self.t_max if end_frame is None else end_frame
+        channel_indices = list(range(n_channels)) if channel_indices is None else channel_indices
+
+        if self.parent_recording is not None:
+            traces = self.parent_recording.get_traces(start_frame, end_frame, channel_indices)
+        else:
+            traces = np.zeros([end_frame - start_frame, n_channels], dtype=self.dtype)
+        
+
+        start = np.searchsorted(self.spike_vector['sample_ind'], start_frame - self.templates.shape[1], side="left")
+        end   = np.searchsorted(self.spike_vector['sample_ind'], end_frame   + self.templates.shape[1], side="right")
+        spike_vector = self.spike_vector[start : end]
+
+        for spike in spike_vector:
+            t = spike['sample_ind']
+            unit_ind = spike['unit_ind']
+            template = self.templates[unit_ind, :]
+
+            start_traces = t - self.nbefore[unit_ind] - start_frame
+            end_traces = start_traces + template.shape[0]
+            if start_traces >= end_frame-start_frame or end_traces <= 0:
+                continue
+
+            start_template = 0
+            end_template = template.shape[0]
+
+            if start_traces < 0:
+                start_template = -start_traces
+                start_traces = 0
+            if end_traces > end_frame - start_frame:
+                end_template = template.shape[0] + end_frame - start_frame - end_traces
+                end_traces = end_frame - start_frame
+
+            traces[start_traces : end_traces] += (template[start_template : end_template].astype(np.float64) * self.amplitude_factor[unit_ind][spike['spike_nb']]).astype(self.dtype)
```

Comment:
Should use the dtype of `traces` otherwise `+=` could fail:

```suggestion
            traces[start_traces : end_traces] += (template[start_template : end_template].astype(np.float64) * self.amplitude_factor[unit_ind][spike['spike_nb']]).astype(traces.dtype)
```

Reply:
Done.